tests/components/pilight/test_init.py

```diff
@@ -66,351 +61,340 @@ def set_callback(self, function):
 
 
-@pytest.mark.skip("Flaky")
-class TestPilight(unittest.TestCase):
-    """Test the Pilight component."""
-
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.skip_teardown_stop = False
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        if not self.skip_teardown_stop:
-            self.hass.stop()
-
-    @patch("homeassistant.components.pilight._LOGGER.error")
-    def test_connection_failed_error(self, mock_error):
-        """Try to connect at 127.0.0.1:5001 with socket error."""
-        with assert_setup_component(4):
-            with patch(
-                "pilight.pilight.Client", side_effect=socket.error
-            ) as mock_client:
-                assert not setup_component(
-                    self.hass, pilight.DOMAIN, {pilight.DOMAIN: {}}
-                )
-                mock_client.assert_called_once_with(
-                    host=pilight.DEFAULT_HOST, port=pilight.DEFAULT_PORT
-                )
-                assert mock_error.call_count == 1
-
-    @patch("homeassistant.components.pilight._LOGGER.error")
-    def test_connection_timeout_error(self, mock_error):
-        """Try to connect at 127.0.0.1:5001 with socket timeout."""
-        with assert_setup_component(4):
-            with patch(
-                "pilight.pilight.Client", side_effect=socket.timeout
-            ) as mock_client:
-                assert not setup_component(
-                    self.hass, pilight.DOMAIN, {pilight.DOMAIN: {}}
-                )
-                mock_client.assert_called_once_with(
-                    host=pilight.DEFAULT_HOST, port=pilight.DEFAULT_PORT
-                )
-                assert mock_error.call_count == 1
-
-    @patch("pilight.pilight.Client", PilightDaemonSim)
-    @patch("homeassistant.core._LOGGER.error")
-    @patch("homeassistant.components.pilight._LOGGER.error")
-    def test_send_code_no_protocol(self, mock_pilight_error, mock_error):
-        """Try to send data without protocol information, should give error."""
-        with assert_setup_component(4):
-            assert setup_component(self.hass, pilight.DOMAIN, {pilight.DOMAIN: {}})
-
-            # Call without protocol info, should be ignored with error
-            self.hass.services.call(
-                pilight.DOMAIN,
-                pilight.SERVICE_NAME,
-                service_data={"noprotocol": "test", "value": 42},
-                blocking=True,
+@patch("homeassistant.components.pilight._LOGGER.error")
+async def test_connection_failed_error(mock_error, hass):
+    """Try to connect at 127.0.0.1:5001 with socket error."""
+    with assert_setup_component(4):
+        with patch("pilight.pilight.Client", side_effect=socket.error) as mock_client:
+            assert not await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)